signin/index.js

```diff
@@ -44,6 +54,7 @@ passwordInput.addEventListener("keypress", function (event) {
   }
```

Comment:
그랬을 때, form의 onsubmit 액션을 이렇게 활용할 수 있을거에요! 
```
/**
   * Asynchronously retrieves user data by email from a server.
   *
   * This function sends a POST request to a server with the user's email in the request body.
   * If the server responds with a successful status, the function parses the JSON response
   * and returns the user data. In case of a server-side error, the server is expected to respond
   * with a JSON object containing `null` data and an error message. The function then throws
   * an error with the received error message.
   *
   * @param {string} email - The email address used to query the user data from the server.
   * @returns {Promise<Object | null>} A promise that resolves to the user data object on successful retrieval.
   * @throws {Error} Throws an error with the error message received from the server.
   */
  async function getUserByEmail(email) {
    const response = await fetch("...", { method: "POST", body: JSON.stringify({ email }) })

    if (!response.ok) {
      const errorResponse = await response.json() // 만약 오류가 발생했고, 서버측에서 { data: null, error: '오류메시지' } 형태로 데이터를 전달하고 있는 경우
      const errorMessage = errorResponse.error // 이런식으로 꺼내와 활용 가능
      throw new Error(errorMessage) // 서버에서 전달한 에러메시지로 catch문에 전달
    }

    const json = await response.json()
    const user = json.data // 만약 이메일에 해당하는 사용자가 있다면, user정보를 반환하고, 이메일에 해당하는 사용자가 없다면 null 반환

    return user
  }


  async function handleLogin(e) {
    e.preventDefault()

    const email = emailInputElement.value
    const password = passwordInputElement.value

    if (isTextEmpty(email)) {
      return showError( ... ) 
    }

    if (isTextEmpty(password)) {
      return showError( ... )
    }

    if (!isValidEmailFormat(email)) {
      return showError( ... )
    }

    try {
      const user = getUserByEmail(email) // getUserByEmail에서 서버 호출도중 오류 발생한다면 catch문으로 전달 

      // user가 null인경우 핸들링
      // 이 때, 등록된 이메일이 없습니다 라고 오류메시지를 전달하기보단, 그냥 이메일 비번 없다고 처리하는게 좀 더 보안상 좋음!
      if (user === null) {
        showError("이메일 정보 확인해주세요")
        showError ("비밀번호 정보 확인해주세요")
        return
      }

      // db에 저장된 비번이랑 사용자가 입력한 비번이 틀린경우
      if (user.password !== password) {
        showError("이메일 정보 확인해주세요")
        showError ("비밀번호 정보 확인해주세요")
        return
      }
      
      // 로그인 후 처리 진행
      window.href = ~~~~ 
    } catch (error) {
      alert("서버 오류가 발생했어요!", error.message)
    }
  }
  signInFormElement.addEventListener("submit", handleLogin)
```